src/App.js

```diff
@@ -28,31 +28,94 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+
+
+  const switchPlayers = () => {
+    if (player === PLAYER_1) {
+      setPlayer(PLAYER_2)
+    }
+    if (player === PLAYER_2) {
+      setPlayer(PLAYER_1)
+    }; 
+  };
+
+ 
+
 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  let onClickCallback = (squareClickedOnId) => {
+    let updatedSquares = []
+    
+    if (checkForWinner() !== null) {
+      return; 
+    }
+    
+    squares.forEach((row) => {
+      let updatedSquareRow = []
+
+      row.forEach((square) => {
+        if (squareClickedOnId === square.id){
+          if (square.value === ''){
+          const updatedSquare = {id: squareClickedOnId, value: player }
+          updatedSquareRow.push(updatedSquare)
+          switchPlayers()
+          } else {
+            updatedSquareRow.push(square) 
+          }
+        }else {
+          updatedSquareRow.push(square)
+        };
+      });
+
+      updatedSquares.push(updatedSquareRow)
+    });
+   
+    setSquares(updatedSquares);
+  };
 
 
   const checkForWinner = () => {
     // Complete in Wave 3
+    for (let i = 0; i < 3; i++) {
+      if (squares[i][0].value !== '' && squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value) {
+        return squares[i][0].value;
+      } else if (squares[0][i].value !== '' && squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value) {
+        return squares[0][i].value;
+      };
+    };
```

Comment:
I enjoy the minimizing of code-writing that comes from putting the row and column checks into one loop. 
That said it does make it slightly less readable. One area for improvement would be separating this part and the diagonal part into multiple new functions for readability.